design/token-mechanics-v2.1.md

```diff
@@ -1,46 +1,55 @@
+- A challenge can be accepted or rejected by the owner of the challenged entry. Regardless of the result, it enters a period of N days during which it can be appealed. If not appealed, it is then confirmed and executed. Appeals allow challengers to have their rejected challengers reviewed, and prevent dishonest developers from colluding with a challenger to drain funds from the other vouchers of an entry.
+- Any user can appeal a resolution of a challenge (during the appealing period) by staking tokens behind it. Resolution of appeals will be centralized initially, and then solved via a PLCR voting by all ZEP token holders.
+
+## Issues
+
+1- How we can avoid users unvouching their tokens given a new challenge yet unsolved?
+  
+  We can lock all funds that are needed to pay for all open challenges, and allow users to unvouch the rest. Note that this may not be implementable due to gas constraints. Another simple option is to outright lock all funds on a challenged entry. 
+  
+  We may also allow users to signal their willingness to retreat from a package, in which case new challenges do not apply to them, and once the last challenge to their funds ends, then they are allowed to retreat.ges
+
+2- How can we prevent a user from grabbing an off-chain challenge and submitting it under their name before the original auditor?
+
+  We can handle front-running scenarios by manually checking both challenges and deciding which is the real one based on off-chain information.
 
-4- What would happen if package registrars' vouch reaches an amount lower than the minimum stake due to challenges payout?
-> Users will have to provide a certain amount of tokens the next time they register a version to reach the minimum stake.
-  We should make sure this is an edge scenario since we will be forcing package registrars to buy more tokens.
+3- How we can make sure challenges payout cannot be manipulated based on their resolution order? 
+  
+  An obvious approach would be to solve challenges in a FIFO order. Note that this blocks solving a challenge until all previous ones have been solved as well. A challenge will then _lock_ a certain amount of tokens to make sure it can be paid, and new challenges will then take a percentage of the _free_ tokens on an entry.
+  
+4- How we can make sure there are enough tokens to pay successful challenges?
+
+  If every challenge takes a percentage of the _free_ tokens of an entry, then we should always have enough tokens to pay, even if the payout for challenges that occur later in time is lower.
+
+5- What would happen if the tokens vouched reaches an amount lower than the minimum stake due to challenges payout?
+
+  We can leave the min stake out of the number of _free_ tokens. This way, the min stake is exclusively used for spam prevention, and not for paying out challenges.
```

Comment:
In that case, an entry will need to have at least something vouched on it to be able to receive a challenge right?